lib/instrumentation/core/http-outbound.js

```diff
@@ -138,6 +138,8 @@ function instrumentRequest(agent, opts, makeRequest, hostname, segment) {
   const transaction = segment.transaction
   const outboundHeaders = Object.create(null)
 
+  opts.headers = opts.headers || {};
```

Comment:
Can you please write a unit test for this?

Here's one that exhibits the issue.

```js
  t.test('should not crash when req.headers is null', (t) => {
    const req = new events.EventEmitter()
    helper.runInTransaction(agent, function (transaction) {
      const path = '/asdf'
      const name = NAMES.EXTERNAL.PREFIX + HOSTNAME + ':' + PORT + path

      instrumentOutbound(agent, { headers: null, host: HOSTNAME, port: PORT }, makeFakeRequest)
      t.equal(transaction.trace.root.children[0].name, name)

      function makeFakeRequest() {
        req.path = '/asdf?a=b&another=yourself&thing&grownup=true'
        return req
      }
    })
    t.end()
  })
```

You should put it in `/test/unit/instrumentation/http/outbound.test.js`

Reply:
Thanks for this! I had actually looked for a test to update, but was looking in `/test/unit/instrumentation/core` and didn't see one. I added the test and will @ you once I see things have passed.

Reply:
I tweaked it slightly to be more specific. I didn't mean for you to copy/paste but use for inspiration.  Anyways, we appreciate the fix 🙏🏻

Reply:
Oh, sorry! Haha. Thanks for the tweak. I should have paid more attention to what you were proposing.